Formula/y/yazi.rb

```diff
@@ -0,0 +1,28 @@
+class Yazi < Formula
+  desc "️Blazing fast terminal file manager written in Rust, based on async I/O"
+  homepage "https://github.com/sxyazi/yazi"
+  url "https://github.com/sxyazi/yazi/archive/refs/tags/v0.1.3.tar.gz"
+  sha256 "e6738a12896ff0ab081a306e6334702a93dcb1039e4c596eb6ee723a6bc037cc"
+  license "MIT"
+  head "https://github.com/sxyazi/yazi.git",
+       branch: "main"
+
+  depends_on "rust" => :build
+  depends_on "fd"
+  depends_on "ffmpegthumbnailer"
+  depends_on "fzf"
+  depends_on "jq"
+  depends_on "poppler"
+  depends_on "ripgrep"
+  depends_on "unar"
+  depends_on "zoxide"
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "app")
+    bin.install "./target/release/yazi"
+  end
+
+  test do
+    assert_equal "yazi #{version}", shell_output("#{bin}/yazi -V").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
check out the other terminal file manager examples.